src/module.ts

```diff
@@ -26,13 +26,17 @@ export default defineNuxtModule<Partial<Options>>({
 
     nuxt.hook('vite:extendConfig', (config, { isClient }) => {
       const mode = isClient ? 'client' : 'server'
+      const sourcemapValue = nuxt.options.sourcemap[mode]
+      if (typeof sourcemapValue !== 'boolean') {
+        throw new TypeError(`Expected a boolean for sourcemap[${mode}], but received ${typeof sourcemapValue}.`)
+      }
```

Comment:
`sourcemapValue: boolean | undefined` ?

This way of handling it feels a little redundant

Reply:
I agree.  I will revert this to the original.